pkg/nuctl/command/invoke.go

```diff
@@ -151,6 +174,63 @@ func newInvokeCommandeer(ctx context.Context, rootCommandeer *RootCommandeer) *i
 	return commandeer
 }
 
+func (i *invokeCommandeer) enrichOptionsForExternalIP(invocationURLs []string) error {
+
+	// provided external ip address,
+	if i.externalIPAddresses != "" {
+
+		// function has node port
+		if functionNodePort := i.createFunctionInvocationOptions.
+			FunctionInstance.
+			GetStatus().
+			HTTPPort; functionNodePort != 0 {
+			externalIPAddresses, err := i.rootCommandeer.platform.GetExternalIPAddresses()
+			if err != nil {
+				return errors.Wrap(err, "Failed to get external IP addresses")
+			}
+			i.createFunctionInvocationOptions.URL = fmt.Sprintf("%s:%d", externalIPAddresses[0],
+				functionNodePort)
+			i.createFunctionInvocationOptions.SkipURLValidation = true
+		} else {
+			return errors.New("Function has no node port and thus cannot be invoked externally " +
+				"while providing external ip addresses")
+		}
+		return nil
+	}
+
+	if len(invocationURLs) < 2 {
+		return errors.New("Function has no external invocation url")
+	}
+
+	// use last invocation url
+	// implementation detail: first url is the intra-cluster url, following urls are external urls
+	// the last url is the one that is most likely to be an ingress, if not, node port
+	i.createFunctionInvocationOptions.URL = invocationURLs[len(invocationURLs)-1]
+
+	// replace the host with the external ip address in case running from a container / cluster
+	// in which case that host's ip address is not accessible within the docker network / k8s cluster
+	if common.RunningInContainer() || common.IsInKubernetesCluster() {
+		parsedURL, err := url.Parse(i.createFunctionInvocationOptions.URL)
+		if err != nil {
+			return errors.Wrap(err, "Failed to parse invocation URL")
+		}
+
+		if common.StringSliceContainsString(
+			[]string{"localhost", "0.0.0.0", "127.0.0.1"}, parsedURL.Hostname()) {
+			externalIPAddress, err := i.rootCommandeer.platform.GetExternalIPAddresses()
+			if err != nil {
+				return errors.Wrap(err, "Failed to get external IP addresses")
+			}
+			i.rootCommandeer.loggerInstance.DebugWith("Overriding external IP address",
+				"currentExternalIPAddress", parsedURL.Hostname(),
+				"overridingExternalIPAddress", externalIPAddress)
+			parsedURL.Host = fmt.Sprintf("%s:%s", externalIPAddress[0], parsedURL.Port())
```

Comment:
You want to override the url in `i.createFunctionInvocationOptions.URL` and not the closured variable `parsedURL`.

Reply:
fixed